dvc/repo/__init__.py

```diff
@@ -482,6 +463,25 @@ def open_by_relpath(self, path, remote=None, mode="r", encoding=None):
     def close(self):
         self.scm.close()
 
+    @contextmanager
+    def lock_repo(self):
```

Comment:
Usage in tests indicates that it might be used via `Repo.lock_repo(repo)`, maybe we should not use `self` here, but `repo` instead? It would look similar to what we are doing in many `dvc.repo.*` packages and also indicate that one not necessarily needs an object to call it. I don't have a strong opinion on that.

Reply:
@pared, that's just an easy way to test a `contextmanager`. You can just use it as:
```python:
with repo.lock_repo():
  pass
```

Reply:
I guess I was too fast with that comment, test led me to believe that there might be the use case where you lock repo twice, which will probably not be a prevalent use case, now that I think about that.

Reply:
> where you lock repo twice

@pared, this was meant to be used in the Viewer to prevent calling `_reset()` too many times when calling Repo's top-level APIs.

cc @efiop let me know if we should maintain compatibility for the Viewer.
It made sense to me to move it inside `Repo` as  I see it as a part of the `Repo` itself and that it would get into cyclic import issue. But, looks like it won't get into cyclic import issue.

Reply:
@skshetry Is this change needed for this PR?

Reply:
> Is this change needed for this PR?

No, I just find it easier it being in `Repo`, easier to test/mock, and no need to import it at all. And, it depends too much on the private variables anyway. Rolled back for now.